components/inputs/input-date.js

```diff
@@ -155,25 +159,35 @@ class InputDate extends LocalizeStaticMixin(LitElement) {
 		this._dropdown = this.shadowRoot.querySelector('d2l-dropdown-content');
 		this._calendar = this.shadowRoot.querySelector('d2l-calendar');
 
+		this.addEventListener('blur', this._handleBlur);
 		this.addEventListener('d2l-localize-behavior-language-changed', () => {
 			this._dateTimeDescriptor = getDateTimeDescriptorShared(true);
-			this.requestUpdate();
+			this._getContentWidth();
 		});
+
+		this._formattedValue = this.emptyText ? this.emptyText : '';
+		const fontsPromise = document.fonts ? document.fonts.ready : Promise.resolve();
+		await Promise.resolve(fontsPromise);
```

Comment:
Would this not do the same?

```suggestion
		await document.fonts ? document.fonts.ready : Promise.resolve();
```